### PR TITLE
Vision processors - replace FE with IPs

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Replaces feature extractors with image processors in the `Processor` class which bundle together tokenizers and feature extractor. 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/clip/processing_clip.py [hunk 1]
```diff
@@ -15,37 +15,54 @@
...
+    image_processor_class = "CLIPImageProcessor"
     tokenizer_class = ("CLIPTokenizer", "CLIPTokenizerFast")
 
-    def __init__(self, feature_extractor, tokenizer):
-        super().__init__(feature_extractor, tokenizer)
-        self.current_processor = self.feature_extractor
+    def __init__(self, image_processor=None, tokenizer=None, **kwargs):
+        if "feature_extractor" in kwargs:
+            warnings.warn(
+                "The `feature_extractor` argument is deprecated and will be removed in v4.27, use `image_processor`"
```
Comment (sgugger) [inline src/transformers/models/clip/processing_clip.py:45]:
```suggestion
                "The `feature_extractor` argument is deprecated and will be removed in v5, use `image_processor`"
```

File: src/transformers/models/clip/processing_clip.py [hunk 2]
```diff
@@ -109,5 +126,14 @@ def decode(self, *args, **kwargs):
...
         tokenizer_input_names = self.tokenizer.model_input_names
-        feature_extractor_input_names = self.feature_extractor.model_input_names
-        return list(dict.fromkeys(tokenizer_input_names + feature_extractor_input_names))
+        image_processor_input_names = self.image_processor.model_input_names
+        return list(dict.fromkeys(tokenizer_input_names + image_processor_input_names))
+
+    @property
+    def feature_extractor_class(self):
+        warnings.warn(
+            "`feature_extractor_class` is deprecated and will be removed in v4.27. Use `image_processor_class`"
```
Comment (sgugger) [inline src/transformers/models/clip/processing_clip.py:135]:
```suggestion
            "`feature_extractor_class` is deprecated and will be removed in v5. Use `image_processor_class`"
```

**Review by sgugger — COMMENTED** [review]
Thanks! Just one comment on the deprecation warning to propagate to all processors.

**Comment by ydshieh** [comment]
Hi @amyeroberts We still can't change this part at this moment?

```python
class ProcessorMixin(PushToHubMixin):
    """
    This is a mixin used to provide saving/loading functionality for all processor classes.
    """

    attributes = ["feature_extractor", "tokenizer"]
    # Names need to be attr_class for attr in attributes
    feature_extractor_class = None
    tokenizer_class = None
```

File: src/transformers/models/clip/processing_clip.py [hunk 3]
```diff
@@ -109,5 +126,13 @@ def decode(self, *args, **kwargs):
...
+        image_processor_input_names = self.image_processor.model_input_names
+        return list(dict.fromkeys(tokenizer_input_names + image_processor_input_names))
+
+    @property
+    def feature_extractor_class(self):
+        warnings.warn(
+            "`feature_extractor_class` is deprecated and will be removed in v5. Use `image_processor_class` instead.",
+            FutureWarning,
+        )
+        return self.image_processor_class
```
Comment (ydshieh) [inline src/transformers/models/clip/processing_clip.py:138]: Should we also allow to give `feature_extractor` ?

**Review by sgugger — APPROVED** [review]

File: src/transformers/models/clip/processing_clip.py (hunk 3 above)
Comment (ydshieh) [reply src/transformers/models/clip/processing_clip.py:138]: I mean adding a  property `feature_extractor` too ?

Comment (amyeroberts) [reply src/transformers/models/clip/processing_clip.py:138]: Yes, that's a good point. I'll add that too.

Comment (amyeroberts) [reply src/transformers/models/clip/processing_clip.py:138]: I've added a `feature_extractor` class to all the processors. Let me know if that matches what you were thinking :)

**Review by ydshieh — APPROVED** [review]
Thanks @amyeroberts !

**Pull request merged**